src/datastore/pg-write-store.ts

```diff
@@ -450,6 +450,10 @@ export class PgWriteStore extends PgStore {
         }
       }
 
+      if (!this.isEventReplay) {
+        await this.reconcileMempoolStatus(sql);
+      }
+
       if (!this.isEventReplay) {
```

Comment:
Could you consolidate these two code blocks into the same conditional? Same in the other usage